lib/core/test/unit/Cardano/Wallet/Primitive/CoinSelection/MA/RoundRobinSpec.hs

```diff
@@ -371,30 +453,72 @@ prop_runSelection_UTxO_notEnough (Small index) = monadicIO $ do
     balanceRequested = adjustAllQuantities (* 2) balanceAvailable
 
 prop_runSelection_UTxO_exactlyEnough
-    :: Small UTxOIndex -> Property
+    :: Small UTxOIndex
+    -> Property
 prop_runSelection_UTxO_exactlyEnough (Small index) = monadicIO $ do
     SelectionState {selected, leftover} <-
-        run $ runSelection index balanceRequested
+        run $ runSelection Nothing index balanceRequested
     let balanceSelected = view #balance selected
     let balanceLeftover = view #balance leftover
     assert $ balanceSelected == balanceRequested
     assert $ balanceLeftover == TokenBundle.empty
   where
     balanceRequested = view #balance index
 
+prop_runSelection_UTxO_extraSourceUsed
+    :: Maybe Coin
+    -> Small UTxOIndex
+    -> Property
+prop_runSelection_UTxO_extraSourceUsed extraSource (Small index) = do
+    let hasSomeAda = maybe False (/= TokenBundle.empty) almostEverything
```

Comment:
Asserting that a `TokenBundle` is non-empty is unfortunately not enough to guarantee that it has some ada. We must check the `Coin` value directly.

I made a small fix in: 781fc4809b892eb11f8b30751ab2ee09bc8acd0f. Feel free to cherry-pick it if it works for you (or just copy and paste if you prefer).